io/src/main/java/software/amazon/smithy/java/io/datastream/InputStreamDataStream.java

```diff
@@ -27,17 +21,12 @@ final class InputStreamDataStream implements DataStream {
     }
 
     @Override
-    public ByteBuffer waitForByteBuffer() {
-        try {
-            return ByteBuffer.wrap(inputStream.readAllBytes());
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+    public InputStream asInputStream() {
+        if (consumed) {
```

Comment:
FYI, this isn't thread safe, I am not entirely sure if that's acceptable here.

Reply:
I figured that InputStream and reading data is entirely not thread safe, so it was fine